Add zdcreco to Run3 Forests

matt2275 opened the pull request:
#### PR description:

Added zdcrecoRun3 and new zdcrechit analyzer to Run3 forests for HI and PPRef

#### PR validation:

Tested both `forest_miniAOD_run3_DATA.py` and `forest_miniAOD_run3_PPRef_DATA.py` locally
 
`forest_miniAOD_run3_DATA.py`  runs as it without crashing

`forest_miniAOD_run3_PPRef_DATA.py` crashes is using 2023 like in the example both works with 2024 files like /store/data/Run2024J/ZeroBias/MINIAOD/PromptReco-v1/000/387/396/00000/cce7fdf7-b241-41d6-9498-2d9eb9be2863.root

This is an expected crash since the ZDC was not installed during ppref in 2023

matt2275 committed:
Add zdcreco to Run3 Forests

boundino commented:
Thanks! Could you move the producer `zdcrecoRun3` in a _cfi along with the parameters there, and only keep the parameters that general users may change in forest_xx.py? Ideally we hide all the parameters in _cfi and only experts will change them manually when necessary

acturner commented:
Hi - I have been getting these emails for many years now. Could someone help to unsubscribe me? Thanks in advance!On Oct 29, 2024, at 11:12, Jing Wang ***@***.***> wrote:﻿
Thanks! Could you move the producer zdcrecoRun3 in a _cfi along with the parameters there, and only keep the parameters that general users may change in forest_xx.py? Ideally we hide all the parameters in _cfi and only experts will change them manually when necessary

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

matt2275 committed:
Added zdcrecoRun3_cfi

matt2275 commented:
I added the cfi file and updated the forest files.

jusaviin commented:
I am bringing up here what @boundino commented on https://github.com/CmsHI/cmssw/pull/398. You have different definition for the MyZDCDigi struct in ZDCRecHitAnalyzer and ZDCTreeProducer. You should have only one definition for this. If you need different definitions, the structs should have different names. Since this pull request is related, maybe this can be fixed as part of this pull request.

Also, I see that with the new pull request you do not reference the run2 specific classes QWZDC2018Producer and QWZDC2018RecHit anymore. If these are obsolete in the run3 configuration, can you please remove the source files and python configurations?

jusaviin commented:
@acturner You are probably watching the repository https://github.com/CmsHI/cmssw. You should stop getting the e-mails if you unwatch it in GitHub.

jusaviin commented on HeavyIonsAnalysis/Configuration/test/forest_miniAOD_run3_ppref_DATA.py:
```diff
@@ -44,7 +44,7 @@
 #https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/132X_dataRun3_Prompt_v3
 from Configuration.AlCa.GlobalTag import GlobalTag
-process.GlobalTag = GlobalTag(process.GlobalTag, '141X_dataRun3_Prompt_v3', '')
+process.GlobalTag = GlobalTag(process.GlobalTag, '132X_dataRun3_Prompt_v3', '')
```
Please keep the latest global tag. This was updated in a previous pull request.

matt2275 commented:
@boundino mentions she planned to add a new header file that defined the number of Timeslices and size of the zdcdigis to avoid the definition mismatch. If removing the QWZDC producers is the preferred solution, I can do that instead.

That's for noticing the global tag issue. I forgot to change it back after I finished testing something. GT should be updated now

matt2275 committed:
Fix Global Tag

jusaviin commented:
If @boundino is working on definition mismatch, that is fine. The QWZCD removal is separate from this. If these classes are not used anymore in the new run3 configuration, it is better to remove them from the repository to keep it cleaner and easier to find the relevant files.

boundino commented:
Yes I'm working on the definition mismatch, and will make PR after this one is merged

jusaviin commented:
Great. Can you confirm @matt2275 that QWZDC classes are not needed in the new run3 configuration? If they are not needed, I will merge after they are removed from src and python folders. If they are still needed, I can merge now. We should not leave them there just in case as the classes will be available in commit history and in other branches if for some reason they are useful in the future.

boundino commented:
Also, I realized that ZDCTreeProducer.cc and ZDCRecHitAnalyzer.cc are actually doing the same thing. I'll do a comparison to see if we should just keep one tree producer, and update the names of the overlapped branches so that it's transparent to users. Will double check with ZDC people before make PR.

jusaviin committed:
Merge branch 'forest_CMSSW_14_1_X' into UpdateForests

matt2275 committed:
Remove Old QWZDC files and temp bug fix

Removed temporary fix to the zdcreco saturation and tdc_time.

Also removed old QW ZDC files and updated ZDCTreeProducer to use ZDCHardCodeHelper instead of QWZDCHelper to get hardcoded adc2fc values

matt2275 committed:
Merge branch 'UpdateForests' of https://github.com/matt2275/cmssw into UpdateForests
